src/migrations/20230802000000-internal-goal-duplicate-cleanup.js

```diff
@@ -0,0 +1,2438 @@
+          "xZALActivityReportGoals" AS ( SELECT 'ActivityReportGoals' AS "table", * FROM "ZALActivityReportGoals" z WHERE session_sig = '${sessionSig}'  AND dml_by = -1),
+          "xZALActivityReportGoalResources" AS ( SELECT 'ActivityReportGoalResources' AS "table", * FROM "ZALActivityReportGoalResources" z WHERE session_sig = '${sessionSig}'  AND dml_by = -1),
+          "xZALObjectives" AS ( SELECT 'Objectives' AS "table", * FROM "ZALObjectives" z WHERE session_sig = '${sessionSig}'  AND dml_by = -1),
+          "xZALObjectiveFiles" AS ( SELECT 'ObjectiveFiles' AS "table", * FROM "ZALObjectiveFiles" z WHERE session_sig = '${sessionSig}'  AND dml_by = -1),
+          "xZALObjectiveResources" AS ( SELECT 'ObjectiveResources' AS "table", * FROM "ZALObjectiveResources" z WHERE session_sig = '${sessionSig}'  AND dml_by = -1),
+          "xZALObjectiveTopics" AS ( SELECT 'ObjectiveTopics' AS "table", * FROM "ZALObjectiveTopics" z WHERE session_sig = '${sessionSig}'  AND dml_by = -1),
+          "xZALActivityReportObjectives" AS ( SELECT 'ActivityReportObjectives' AS "table", * FROM "ZALActivityReportObjectives" z WHERE session_sig = '${sessionSig}'  AND dml_by = -1),
+          "xZALActivityReportObjectiveFiles" AS ( SELECT 'ActivityReportObjectiveFiles' AS "table", * FROM "ZALActivityReportObjectiveFiles" z WHERE session_sig = '${sessionSig}'  AND dml_by = -1),
+          "xZALActivityReportObjectiveResources" AS ( SELECT 'ActivityReportObjectiveResources' AS "table", * FROM "ZALActivityReportObjectiveResources" z WHERE session_sig = '${sessionSig}'  AND dml_by = -1),
+          "xZALActivityReportObjectiveTopics" AS ( SELECT 'ActivityReportObjectiveTopics' AS "table", * FROM "ZALActivityReportObjectiveTopics" z WHERE session_sig = '${sessionSig}'  AND dml_by = -1),
+          logs AS (
+            SELECT * FROM "xZALGoals" UNION
+            SELECT * FROM "xZALActivityReportGoals" UNION
+            SELECT * FROM "xZALActivityReportGoalResources" UNION
+            SELECT * FROM "xZALObjectives" UNION
+            SELECT * FROM "xZALObjectiveFiles" UNION
+            SELECT * FROM "xZALObjectiveResources" UNION
+            SELECT * FROM "xZALObjectiveTopics" UNION
+            SELECT * FROM "xZALActivityReportObjectives" UNION
+            SELECT * FROM "xZALActivityReportObjectiveFiles" UNION
+            SELECT * FROM "xZALActivityReportObjectiveResources" UNION
+            SELECT * FROM "xZALActivityReportObjectiveTopics"
+          )
+          SELECT "descriptor", ARRAY_AGG(DISTINCT "table") "tables"
+          FROM "logs" z
+          JOIN "ZADescriptor" zd
+          ON z.descriptor_id = zd.id
+          GROUP BY 1
+          HAVING COUNT(DISTINCT "table") > 1
+          ORDER BY 1,2;
+          `, { raw: true, type: queryInterface.sequelize.QueryTypes.SELECT, transaction });
+
+        if (Array.isArray(cascade) && cascade.length > 0) {
+          throw new Error('Cascade error, unexpected entires found:', cascade);
```

Comment:
This rolls back the transaction, correct? If not, there's also a way to make a sql error occur that rolls back the transaction if there are any records in `cascade`.